tests/unit/helpers/test_search.py

```diff
@@ -26,6 +26,37 @@ def test_init():
         assert param_query[par] == add_params[par]
 
 
+def test_init_legacy_deprecation_warning():
+    with pytest.warns(
+        DeprecationWarning,
+        match="'SearchQuery' is a deprecated name. Use 'SearchQueryV1' instead.",
+    ):
+        SearchQuery()
+
+
+def test_init_v1():
+    query = SearchQueryV1()
+
+    # ensure the version is set to query#1.0.0
+    assert query["@version"] == "query#1.0.0"
+
+    # ensure key attributes initialize to empty lists
+    for attribute in ["facets", "filters", "post_facet_filters", "sorts", "boosts"]:
+        assert query[attribute] == []
+
+    # init with supported fields
+    params = {"q": "foo", "limit": 10, "offset": 0, "advanced": False}
+    param_query = SearchQueryV1(**params)
+    for par in params:
+        assert param_query[par] == params[par]
+
+    # init with additional_fields
+    add_params = {"param1": "value1", "param2": "value2"}
+    param_query = SearchQueryV1(additional_fields=add_params)
+    for par in add_params:
+        assert param_query[par] == add_params[par]
```

Comment:
A test for the initializer itself is good, but I would like us to mirror this with a functional test as well -- one which builds a `SearchQueryV1` document, sends it, and then fetches the last request to verify the payload contents.
Such a test is a good way to confirm that `MISSING` values are set in all of the appropriate slots.

i.e. I believe that the ideal is that we get this:
```python
test_client.post_search(SearchQueryV1(q="*"))
last_req = get_last_request()
parsed_body = json.loads(last_req.body)
assert parsed_body == {"q": "*"}
```